### PR TITLE
[ConnectedKubernetes] Update connected kubernetes api version 2021-10-01

### PR DESCRIPTION
**Pull request by lijinpei2008**
Update connected kubernetes api version2021-10-01

Design doc: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1064

### CODE REVIEW

**Comment by isra-fel** [comment]
Let me move the pull request to draft state. Please feel free to mark it "Ready for review" when you feel comfortable. Thanks

**Comment by lijinpei2008** [comment]
> Let me move the pull request to draft state. Please feel free to mark it "Ready for review" when you feel comfortable. Thanks

Please re-open this PR, check it and merge it.
Thanks.

File: src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1 [hunk 1]
```diff
@@ -149,8 +187,8 @@ function New-AzConnectedKubernetes {
         $PSBoundParameters.Add('IdentityType', $IdentityType)
 
         #Region check helm install
+        $HelmVersion = helm version --short --kubeconfig $KubeConfig
         try {
-            $HelmVersion = helm version --short --kubeconfig $KubeConfig
```
Comment (isra-fel) [inline src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1:153]: Why move this line out of try block?

File: src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1 [hunk 2]
```diff
@@ -230,7 +268,11 @@ function New-AzConnectedKubernetes {
         Set-Item -Path Env:HELM_EXPERIMENTAL_OCI -Value 1
         #Region pull helm chart
         try {
-            helm chart pull $RegisteryPath --kubeconfig $KubeConfig --kube-context $KubeContext
+             if ($HelmVersion -ge "v3.7") {
```
Comment (isra-fel) [inline src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1:271]: Comparing versions as strings is dangerous. Consider if $HelmVersion = "v3.10", the result would be false, but it should be true.

I recommend use .NET type System.Version, for example:
```powershell
[System.Version]"3.10" -gt [System.Version]"3.7"
```

File: src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1 [hunk 3]
```diff
@@ -245,7 +287,11 @@ function New-AzConnectedKubernetes {
             $ChartExportPath = Join-Path -Path $Home -ChildPath '.azure' | Join-Path -ChildPath 'AzureArcCharts'
         }
         try {
-            helm chart export $RegisteryPath --kubeconfig $KubeConfig --kube-context $KubeContext --destination $ChartExportPath
+             if ($HelmVersion -ge "v3.7") {
```
Comment (isra-fel) [inline src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1:290]: same here

File: src/ConnectedKubernetes/custom/Update-AzConnectedKubernetes.ps1 [hunk 4]
```diff
@@ -70,7 +84,13 @@ param(
 
     [Parameter()]
     [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Category('Body')]
-    [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Runtime.Info(PossibleTypes=([Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20210301.IConnectedClusterPatchTags]))]
+    [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.IAny]
```
Comment (isra-fel) [inline src/ConnectedKubernetes/custom/Update-AzConnectedKubernetes.ps1:87]: As we discussed, we are going to change the type to a hashtable

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1 (hunk 1 above)
Comment (lijinpei2008) [reply src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1:153]: In order to be able to use this variable in another method block(Line 233, 248), I moved it out, which now seems dangerous.
I will correct the error and retrieve its value in another method block.

File: src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1 (hunk 2 above)
Comment (lijinpei2008) [reply src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1:271]: Looks great!

Comment (lijinpei2008) [reply src/ConnectedKubernetes/custom/New-AzConnectedKubernetes.ps1]: Done.

**Comment by isra-fel** [comment]
Have you tried to use the `-Property` parmeter?

**Comment by lijinpei2008** [comment]
> Have you tried to use the `-Property` parmeter?

The parameter is read-only, we hid it.

**Review by isra-fel — APPROVED** [review]
LGTM
@lijinpei2008  could you write something that I can put into changelog? Like "Added xxx feature; Support xxx function". Thanks

**Comment by lijinpei2008** [comment]
Create new cmdlet: Get-AzConnectedKubernetesUserCredential.
Update the New-AzConnectedKubernetes to support helm from version 3.7+

**Comment by lijinpei2008** [comment]
> LGTM @lijinpei2008 could you write something that I can put into changelog? Like "Added xxx feature; Support xxx function". Thanks

Create new cmdlet: Get-AzConnectedKubernetesUserCredential.
Update the New-AzConnectedKubernetes to support helm from version 3.7+

**Pull request merged**